pkg/controllers/clusterresourceplacement/clusterresourceplacement_controller_test.go

```diff
@@ -887,3 +889,268 @@ func TestHandleUpdate_failure(t *testing.T) {
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "another-crp-1",
+						Labels: map[string]string{
+							fleetv1beta1.PolicyIndexLabel:      "1",
+							fleetv1beta1.IsLatestSnapshotLabel: "true",
+							fleetv1beta1.CRPTrackingLabel:      "another-crp",
+						},
+					},
+				},
+			},
+			resourceSnapshots: []fleetv1beta1.ClusterResourceSnapshot{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "another-crp-1",
+						Labels: map[string]string{
+							fleetv1beta1.ResourceIndexLabel:    "1",
+							fleetv1beta1.IsLatestSnapshotLabel: "true",
+							fleetv1beta1.CRPTrackingLabel:      "another-crp",
+						},
+						Annotations: map[string]string{
+							fleetv1beta1.ResourceGroupHashAnnotation: "abc",
+						},
+					},
+				},
+			},
+			wantPolicySnapshots: []fleetv1beta1.ClusterSchedulingPolicySnapshot{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "another-crp-1",
+						Labels: map[string]string{
+							fleetv1beta1.PolicyIndexLabel:      "1",
+							fleetv1beta1.IsLatestSnapshotLabel: "true",
+							fleetv1beta1.CRPTrackingLabel:      "another-crp",
+						},
+					},
+				},
+			},
+			wantResourceSnapshots: []fleetv1beta1.ClusterResourceSnapshot{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "another-crp-1",
+						Labels: map[string]string{
+							fleetv1beta1.ResourceIndexLabel:    "1",
+							fleetv1beta1.IsLatestSnapshotLabel: "true",
+							fleetv1beta1.CRPTrackingLabel:      "another-crp",
+						},
+						Annotations: map[string]string{
+							fleetv1beta1.ResourceGroupHashAnnotation: "abc",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			crp := clusterResourcePlacementForTest()
+			crp.Finalizers = []string{fleetv1beta1.ClusterResourcePlacementCleanupFinalizer}
+			now := metav1.Now()
+			crp.DeletionTimestamp = &now
+			objects := []client.Object{crp}
+			for i := range tc.policySnapshots {
+				objects = append(objects, &tc.policySnapshots[i])
+			}
+			for i := range tc.resourceSnapshots {
+				objects = append(objects, &tc.resourceSnapshots[i])
+			}
+			scheme := serviceScheme(t)
+			fakeClient := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(objects...).
+				Build()
+			r := Reconciler{Client: fakeClient, Scheme: scheme, UncachedReader: fakeClient}
+			got, err := r.handleDelete(ctx, crp)
+			if err != nil {
+				t.Fatalf("failed to handle delete: %v", err)
+			}
+			want := ctrl.Result{}
+			if !cmp.Equal(got, want) {
+				t.Errorf("handleDelete() = %+v, want %+v", got, want)
+			}
+			clusterPolicySnapshotList := &fleetv1beta1.ClusterSchedulingPolicySnapshotList{}
+			if err := fakeClient.List(ctx, clusterPolicySnapshotList); err != nil {
+				t.Fatalf("clusterPolicySnapshot List() got error %v, want no error", err)
+			}
+			options := []cmp.Option{
+				cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
```

Comment:
This might be a very commonly used option, should we move it as a top-level variable?